### ISSUE TITLE
tests/image: Use osbuild-composer + updates repo

### ISSUE THREAD

**Issue by atodorov** [osbuild/osbuild-composer#350]
From the discussion in #337 :
- https://github.com/osbuild/osbuild-composer/issues/337#issuecomment-600127039
- https://github.com/osbuild/osbuild-composer/issues/337#issuecomment-600268805
- https://github.com/osbuild/osbuild-composer/issues/337#issuecomment-600521038

Image tests are using `osbuild` instead of `osbuild-composer`. The stated reason is b/c it is not possible to remove the updates repository (can't remove system repos by design) and there is concern by @teg and @ondrejbudai that the updates repository will cause the tests to fail more often. I personally don't see this as such an important problem but I'm not convinced we have enough data to back either of these opinions.

Not using `osbuild-composer` is jumping over how customers will be building these images and we want our tests to be as close to reality as possible.

Opening this issue to collect feedback on $subject and b/c this is independent of the work required to upload & boot into a cloud environment.

**Comment by teg** [comment]
I discussed this with @ondrejbudai, and we agree that the aim is to use 'osbulid-composer` to build the images, we do however have some things we need to overcome before we are able to, so we may not get there immediately.

 - We are currently running the image tests in our CI, but we are not yet running the integration tests. Before we can build images through `osbuild-composer` we need to hook our CI up to run our integration tests and not just our unit tests (we cannot run the composer service from our unit tests, as it needs system integration).
 - As discussed on another issue, it is important that we can specify exactly what repository (and what version of the repository) our tests are run against, so we can verify that all the properties of our images (even the rpm versions) are not changed unintentionally.

We intend to solve both these issues in the near term, and we have ideas for how to do so. And then we intend to move to building images through `osbuild-composer` on top.

**Comment by teg** [comment]
Update: the first of the two items have been solved, but we still do not have the ability to easily specify which repos to build from through the weldr API (they are hard-coded to the official ones).
Thinking more about this, I am wondering if it would not be better to keep the separation the way it is currently done and verify each part of the stack separately. I.e.,

 - One test that verifies the weldr API (`osbulid-test-weldr`)
 - One test that verifies the `blueprint` -> `manifest` translation (currently done in unit tests)
 - One that verifies the `manifest` -> image generation / uploading / booting

We do still have the one image build being done in the `composer-cli` tests, to check that the whole stack does fit together, but ideally each step should be tested thoroughly by itself so that is just a sanity check on top.

What do you guys think?

**Comment by atodorov** [comment]
> but we still do not have the ability to easily specify which repos to build from through the weldr API (they are hard-coded to the official ones).

There is the ability to add additional sources but you can't disable the default ones which come with the RPM (unlike lorax-composer where you can). IDK if we want to rethink this or not.

About the current test approach listed: I think this is fine. Back when this issues was opened there weren't that much tests, if I'm not mistaken there were no cli tests either (even for sanity). 

We do have image builds via composer-cli tests in the lorax repository as well. 

At this time there is still some value in trying to build all image types via the cli for sanity's sake but I'm not sure if this value is worth the effort.  If there are no strong objections then we can close this issue.

**Comment by teg** [comment]
Thanks for the follow-up. When we have the testing infrastructure in a shape we want it, I would like to reconsider adding more image builds, but my assumption is that it will take too long, for too little gain.
I'll close for now.
